v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/dialectadapter/DialectAdapterTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.dialectadapter;
+
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.teleport.v2.source.reader.io.cassandra.iowrapper.CassandraDataSource;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.dialectadapter.mysql.MysqlDialectAdapter;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.dialectadapter.mysql.MysqlDialectAdapter.MySqlVersion;
+import com.google.cloud.teleport.v2.source.reader.io.schema.SourceSchemaReference;
+import com.google.cloud.teleport.v2.source.reader.io.schema.sources.CassandraSchemaReference;
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DialectAdapterTest {
```

Comment:
nit: Please call this JDBCDialectAdapterTest

Reply:
The class id called `DialectAdapter` but it's within the `jdbc` package. Just checking if adding `JDBC` might be redundant?
If we add `JDBC` FQN will be `com.google.cloud.teleport.v2.source.reader.io.jdbc.dialectadapter.JdbcDialectAdapter`
I am fine either ways.
Just a note, naming the main class `JdbcDialectAdapter` will lead to many renamings within the jdbc sub directories.
In case we want to do that, I can take it up as a separate TODO.

Reply:
Reason for this comment is to identify tests quickly from the test name rather than looking up the package name which does not always show up. not a blocker.

Reply:
Thanks. Makes sense. I will name both the main class and test class as `JdbcDialectAdapter` in a separate PR.
For All new classes I will append the source name, for example `CassandraSchemaDiscovery` to avoid this confusion.